Sources/ClientRuntime/Networking/Http/Middlewares/DeserializeMiddleware.swift

```diff
@@ -27,9 +27,8 @@ public struct DeserializeMiddleware<Output: HttpResponseBinding,
                 // error deserialization reads the stream multiple times to first deserialize the protocol error
                 // eg. [RestJSONError](https://github.com/awslabs/aws-sdk-swift/blob/d1d18eefb7457ed27d416b372573a1f815004eb1/Sources/Core/AWSClientRuntime/Protocols/RestJSON/RestJSONError.swift#L38,
                 // and then the service error eg. [AccountNotFoundException](https://github.com/awslabs/aws-sdk-swift/blob/d1d18eefb7457ed27d416b372573a1f815004eb1/Sources/Services/AWSCloudTrail/models/Models.swift#L62)
-                if case let .stream(stream) = copiedResponse.httpResponse.body, !stream.isSeekable {
-                    copiedResponse.httpResponse.body = .stream(CachingStream(base: stream))
-                }
+                let bodyData = try await copiedResponse.httpResponse.body.readAllData()
+                copiedResponse.httpResponse.body = .data(bodyData)
```

Comment:
This code does the same as the previous code did, which is to read the body & save it in-memory so it can be used again.  But here it is done without needing the `CachingStream` type to do it, which was complex & not totally thread safe, and is eliminated in this PR.